sdk/inc/azure/iot/az_iot_provisioning_client.h

```diff
@@ -228,12 +244,8 @@ typedef struct
                          * (which may require several requests) is available only through
                          * #operation_status.  */
   az_span operation_id; /**< Operation ID of the register operation. */
-  az_span operation_status; /**< An #az_span containing the status of the register operation.
-                             * @details This can be one of the following: `unassigned`,
-                             * `assigning`, `assigned`, `failed`, `disabled`.
-                             * az_iot_provisioning_client_parse_operation_status() can optionally
-                             * be used to convert this into
-                             * the #az_iot_provisioning_client_operation_status enum. */
+  az_iot_provisioning_client_operation_status
+      operation_status; /**< An #az_iot_provisioning_client_operation_status of the register operation. */

```

Comment:
nit: Avoid duplicated/redundant info in docs (such as type name, especially since its a long one), since that is generally already visible:
```suggestion
      operation_status; /**< The status of the register operation. */

```